Sources/SwiftDocC/Infrastructure/Link Resolution/LinkResolutionMigration.swift

```diff
@@ -0,0 +1,80 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A scope of configurations for how the documentation context should resolve links while migrating from one implementation to another.
+///
+/// If any reporting is enabled, the documentation context will setup both link resolver to compare them.
+///
+///> Note: This is a temporary configuration that will go away along with the ``DocumentationCacheBasedLinkResolver`` at some point in the future.
+enum LinkResolutionMigrationConfiguration {
+    
+    /// Whether or not the context should the a ``PathHierarchyBasedLinkResolver`` to resolve links.
+    static var shouldUseHierarchyBasedLinkResolver: Bool = {
+        return UserDefaults.standard.bool(forKey: "DocCUseHierarchyBasedLinkResolver")
+            || ProcessInfo.processInfo.environment["DOCC_USE_HIERARCHY_BASED_LINK_RESOLVER"] == "YES"
+    }()
+    
+    /// Whether or not the context should set up a ``PathHierarchyBasedLinkResolver``.
+    ///
+    /// > Node: Check ``shouldUseHierarchyBasedLinkResolver`` to determine which implementation to use to resolve links.
+    static var shouldSetUpHierarchyBasedLinkResolver: Bool {
+        return shouldUseHierarchyBasedLinkResolver || isReportingEnabled
+    }
+    
+    /// Whether or not the context should fully set up a ``DocumentationCacheBasedLinkResolver``.
+    ///
+    /// > Node: Check ``shouldUseHierarchyBasedLinkResolver`` to determine which implementation to use to resolve links.
+    static var shouldFullySetUpCacheBasedLinkResolver: Bool {
```

Comment:
It's not clear to me what 'fully' means in this context, can you explain?

Reply:
This property is no longer needed.